Assets/Scripts/Framework/Gameplay/LaneSystem/Lane.cs

```diff
@@ -0,0 +1,93 @@
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections.Generic;
+
+using ProefExamen.Framework.Gameplay.Values;
+using ProefExamen.Framework.Utils.Libraries.LaneUtils;
+
+namespace ProefExamen.Framework.Gameplay.LaneSystem
+{
+    /// <summary>
+    /// A class that manages a lane.
+    /// </summary>
+    [RequireComponent(typeof(Button))]
+    public class Lane : MonoBehaviour
+    {
+        [SerializeField]
+        private int _id = -1;
+
+        [SerializeField]
+        private Button _button;
+
+        [SerializeField]
+        private List<GameObject> _notes = new();
+
+        [SerializeField]
+        private Vector3 _initialNotePosition;
+
+        [SerializeField]
+        private Vector3 _targetNotePosition;
+
+        /// <summary>
+        /// Gets the private button variable of the Lane.
+        /// </summary>
+        public Button Button => _button;
+
+        private void Start()
+        {
+            if (_button == null)
+                _button = GetComponent<Button>();
+
+            if (_id != -1)
+                _button.onClick.AddListener(SendButtonPress);
+            else
+                Debug.LogError("Lane with button " + _button.name + " has no assigned ID!");
```

Comment:
I would change it to this
```suggestion
            if (_id == -1)
            {
                Debug.LogError("Lane with button " + _button.name + " has no assigned ID!");
                return;
             }
                
            _button.onClick.AddListener(SendButtonPress);
```